be/src/formats/parquet/scalar_column_reader.cpp

```diff
@@ -426,4 +429,181 @@ Status ScalarColumnReader::fill_dst_column(ColumnPtr& dst, ColumnPtr& src) {
+    if (codes_nullable_column->has_null()) {
+        for (size_t i = 0; i < src->size(); i++) {
+            // if null, we assign dict code 0
+            // null = 0, mask = 0xffffffff
+            // null = 1, mask = 0x00000000
+            uint32_t mask = ~(static_cast<uint32_t>(-null_data_ptr[i]));
+            codes[i] = mask & codes[i];
+        }
+    }
+
+    auto* dst_data_column = down_cast<LowCardDictColumn*>(ColumnHelper::get_data_column(dst.get()));
+    SIMDGather::gather(dst_data_column->get_data().data(), _code_convert_map->data(), codes.data(),
+                       DICT_DECODE_MAX_SIZE, src->size());
+
+    if (dst->is_nullable()) {
+        auto* nullable_codes = down_cast<NullableColumn*>(dict_codes.get());
+        auto* nullable_dst = down_cast<NullableColumn*>(dst.get());
+        nullable_dst->null_column_data().swap(nullable_codes->null_column_data());
+        nullable_dst->set_has_null(nullable_codes->has_null());
+    }
+
+    src->reset_column();
+    src = _ori_column;
+
+    return Status::OK();
+}
+
+Status LowCardColumnReader::_check_current_dict() {
+    std::vector<int16_t> code_convert_map;
+
+    // create dict value chunk for evaluation.
+    ColumnPtr dict_value_column = ColumnHelper::create_column(TypeDescriptor(TYPE_VARCHAR), true);
+    RETURN_IF_ERROR(_reader->get_dict_values(dict_value_column.get()));
+
+    size_t dict_size = dict_value_column->size();
+
+    code_convert_map.resize(dict_size + 2);
+    std::fill(code_convert_map.begin(), code_convert_map.end(), 0);
+    auto* local_to_global = code_convert_map.data();
+
+    auto viewer = ColumnViewer<TYPE_VARCHAR>(dict_value_column);
+
+    for (int i = 0; i < dict_size; ++i) {
+        auto slice = viewer.value(i);
+        auto res = _dict->find(slice);
+        if (res == _dict->end()) {
+            if (slice.size > 0) {
+                // error message format used to extract info, carefully
+                return Status::GlobalDictNotMatch(
+                        fmt::format("SlotId: {}, FileName: {} , file doesn't match global dict. ", _slot_id,
+                                    _opts.file->filename()));
+            }
+        } else {
+            local_to_global[i] = res->second;
+        }
+    }
+
+#ifdef DEBUG
+    std::stringstream ss;
+    ss << "dict mapping: ";
+    for (int i = 0; i < dict_size; ++i) {
+        ss << code_convert_map[i] << " ";
+    }
+    LOG(INFO) << ss.str();
+#endif
+
+    _code_convert_map = std::move(code_convert_map);
+
+    return Status::OK();
+}
+
+// LowRowsColumnReader
+
+Status LowRowsColumnReader::read_range(const Range<uint64_t>& range, const Filter* filter, ColumnPtr& dst) {
+    DCHECK(get_column_parquet_field()->is_nullable ? dst->is_nullable() : true);
+    ColumnContentType content_type = ColumnContentType::VALUE;
+
+    if (_tmp_column == nullptr) {
+        _tmp_column = ColumnHelper::create_column(TypeDescriptor::from_logical_type(TYPE_VARCHAR), true);
+    }
```

Comment:
Use `TYPE_VARCHAR_DESC` instead of `TypeDescriptor::from_logical_type(TYPE_VARCHAR)`  ?

Reply:
will do it.